tests/drivers/adc/adc_api/drivers.adc fails to build on nucleo_h753zi

`./scripts/twister -s tests/drivers/adc/adc_api/drivers.adc -p nucleo_h753zi`

Gets:

```
/home/galak/git/zephyr/drivers/adc/adc_stm32.c: In function 'start_read':
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:444:44: error: 'LL_ADC_OVS_RATIO_2' undeclared (first use in this function); did you mean 'LL_ADC_INJ_RANK_2'?
  444 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_2,
      |                                            ^~~~~~~~~~~~~~~~~~
      |                                            LL_ADC_INJ_RANK_2
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:444:44: note: each undeclared identifier is reported only once for each function it appears in
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:449:44: error: 'LL_ADC_OVS_RATIO_4' undeclared (first use in this function); did you mean 'LL_ADC_INJ_RANK_4'?
  449 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_4,
      |                                            ^~~~~~~~~~~~~~~~~~
      |                                            LL_ADC_INJ_RANK_4
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:454:44: error: 'LL_ADC_OVS_RATIO_8' undeclared (first use in this function); did you mean 'LL_ADC_REG_RANK_8'?
  454 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_8,
      |                                            ^~~~~~~~~~~~~~~~~~
      |                                            LL_ADC_REG_RANK_8
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:459:44: error: 'LL_ADC_OVS_RATIO_16' undeclared (first use in this function); did you mean 'LL_ADC_REG_RANK_16'?
  459 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_16,
      |                                            ^~~~~~~~~~~~~~~~~~~
      |                                            LL_ADC_REG_RANK_16
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:464:44: error: 'LL_ADC_OVS_RATIO_32' undeclared (first use in this function)
  464 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_32,
      |                                            ^~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:469:44: error: 'LL_ADC_OVS_RATIO_64' undeclared (first use in this function); did you mean 'LL_ADC_INJ_RANK_4'?
  469 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_64,
      |                                            ^~~~~~~~~~~~~~~~~~~
      |                                            LL_ADC_INJ_RANK_4
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:474:44: error: 'LL_ADC_OVS_RATIO_128' undeclared (first use in this function); did you mean 'LL_ADC_REG_RANK_12'?
  474 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_128,
      |                                            ^~~~~~~~~~~~~~~~~~~~
      |                                            LL_ADC_REG_RANK_12
/home/galak/git/zephyr/drivers/adc/adc_stm32.c:479:44: error: 'LL_ADC_OVS_RATIO_256' undeclared (first use in this function)
  479 |   LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_256,
      |                                            ^~~~~~~~~~~~~~~~~~~~
```

--- a/drivers/adc/adc_stm32.c
+++ b/drivers/adc/adc_stm32.c
@@ -441,42 +441,74 @@ static int start_read(const struct device *dev,
 		break;
 	case 1:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else /* CONFIG_SOC_SERIES_STM32H7X */
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_2,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_1);
 		break;
 	case 2:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else /* CONFIG_SOC_SERIES_STM32H7X */
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_4,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_2);
 		break;
 	case 3:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else /* CONFIG_SOC_SERIES_STM32H7X */
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_8,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_3);
 		break;
 	case 4:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else /* CONFIG_SOC_SERIES_STM32H7X */
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_16,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_4);
 		break;
 	case 5:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else /* CONFIG_SOC_SERIES_STM32H7X */
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_32,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_5);
 		break;
 	case 6:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else /* CONFIG_SOC_SERIES_STM32H7X */
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_64,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_6);
 		break;
 	case 7:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_128,
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 						    LL_ADC_OVS_SHIFT_RIGHT_7);
 		break;
 	case 8:
 		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		LL_ADC_ConfigOverSamplingRatioShift(adc, sequence->oversampling,
+#else
 		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_256,
+#endif
 						    LL_ADC_OVS_SHIFT_RIGHT_8);
 		break;
 	default:

